src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7788,6 +7794,7 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
             foreach (var scope in new ExtensionScopes(this))
             {
                 if (!left.Type.IsTypeParameter()
```

Comment:
>if (!left.Type.IsTypeParameter() [](http://example.com/codeflow?start=16&length=32)

Could you please add comments. Why there are two somewhat similar, but different `if` statements in this loop? I am reading the code and realizing that I have no idea why things are done in this way.  Why type checks are important in the first `if`, but not in the second, etc. It would be good to clear capture the intent and explain how implementation follows/matches it.